### PR TITLE
SearchUI - Add Find Membership using searchkit

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Overview
----------------------------------------
Add Find Membership using searchkit

Find Membership using SK -

Find Memberships

Comments
----------------------------------------
Gitlab: https://lab.civicrm.org/dev/core/-/issues/3912

### CODE REVIEW

File: ext/civi_member/Civi/Api4/Service/Spec/Provider/MembershipGetSpecProvider.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  /**
+   * @param \Civi\Api4\Service\Spec\RequestSpec $spec
+   */
+  public function modifySpec(RequestSpec $spec): void {
+    $spec->addFieldSpec(new FieldSpec('is_primary_member', 'Membership', 'Integer'));
+    $field = (new FieldSpec('is_primary_member', 'Membership', 'Boolean'))
+      ->setTitle(ts('Is Primary Member?'))
+      ->setDescription(ts('Is this a primary membership?'))
+      ->setInputType('Number')
```
Comment (colemanw) [inline ext/civi_member/Civi/Api4/Service/Spec/Provider/MembershipGetSpecProvider.php:34]: Should this be "Radio"?

File: ext/civi_member/Civi/Api4/Service/Spec/Provider/MembershipGetSpecProvider.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * @service
+ * @internal
+ */
+class MembershipGetSpecProvider extends AutoService implements Generic\SpecProviderInterface {
+
+  /**
+   * @param \Civi\Api4\Service\Spec\RequestSpec $spec
+   */
+  public function modifySpec(RequestSpec $spec): void {
+    $spec->addFieldSpec(new FieldSpec('is_primary_member', 'Membership', 'Integer'));
```
Comment (colemanw) [inline ext/civi_member/Civi/Api4/Service/Spec/Provider/MembershipGetSpecProvider.php:30]: Should this be 
```suggestion
    $spec->addFieldSpec(new FieldSpec('is_primary_member', 'Membership', 'Boolean'));
```

**Comment by jitendrapurohit** [comment]
Have updated the required change. Looks good? @colemanw

**Comment by colemanw** [comment]
Looks good @jitendrapurohit.
Juist one thing, It looks like you made changes to an `xml` file so the DAO needs to be regeneated using `bin/setup.sh -g`.

**Pull request merged**